### PR TITLE
Adds a flat routing implementation

### PR DESCRIPTION
**Pull request by nkcr**
This flat routing implementation assumes a N-N connectivity among nodes. It brings more stability on a small topology, as failing tree nodes can break communications.

This is basically a simplified copy of the tree routing implementation.

Also updates `memcoin` to use by default the flat routing. A new flag `routing` can specify either `flat` (default) or `tree`.

### CODE REVIEW

File: mino/router/flat/types/handshake.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+// Deserialize implements serde.Factory. It populates the handshake if
+// appropriate, otherwise it returns an error.
+func (fac HandshakeFactory) Deserialize(ctx serde.Context, data []byte) (serde.Message, error) {
+	return fac.HandshakeOf(ctx, data)
+}
+
+// HandshakeOf implements router.HandshakeFactory. It populates the handshake if
+// appropriate, otherwise it returns an error.
+func (fac HandshakeFactory) HandshakeOf(ctx serde.Context, data []byte) (router.Handshake, error) {
```
Comment (jbsv) [inline mino/router/flat/types/handshake.go:72]: I don't know if we need an extra alias function here

**Review by jbsv — APPROVED** [review]
LGTM. minor comments

File: mino/minogrpc/controller/mod.go [hunk 2]
```diff
@@ -67,6 +67,12 @@ func (m miniController) SetCommands(builder node.Builder) {
 			Value:    "",
 			Required: false,
 		},
+		cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (nkcr) [reply mino/minogrpc/controller/mod.go:70]: We provide working default arguments, which is even shorted than providing a config file. Defaults arguments are set with `Value`.

File: mino/router/flat/mod.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+// NewTable creates a new routing table
+func NewTable(addrs []mino.Address) Table {
+	return Table{
+		known: addrs,
+	}
+}
+
+// Make implements router.RoutingTable. It creates a packet with the source
+// address, the destination addresses and the payload.
+func (t Table) Make(src mino.Address, to []mino.Address, msg []byte) router.Packet {
```
[the remark this answers is not in the document]
Comment (nkcr) [reply mino/router/flat/mod.go:80]: True, `MakePacket` would be better. The routing interface is not very mature anyway and would need to be updated.

**Pull request merged**